### DISCUSSION TITLE
Use of degcor ergm term with target statistics

### DISCUSSION THREAD

**Discussion by smjenness** [statnet/ergm#577] [General]
Hi @krivit : could you provide guidance on how to implement an ergm with target statistics for the `degcor` term. The help documentation for this term is not detailed.

For example, in the following edges-only model, I find that the mean degree correlation, output from `degcor`, is `-0.07810538`.

```r
nw <- network.initialize(1000, directed = FALSE)
g <- ergm(nw ~ edges, target.stats = 1250)
g <- simulate(g, nsim = 1000, monitor = nw ~ degcor, output = "stats")
colMeans(g)
```

Then, trying to fit an ergm with this expected value, I get an error as follows. Is `ergm` expecting a matrix here, despite the model outputting a single correlation value?

```r
g2 <- ergm(nw ~ edges + degcor, target.stats = c(1250, -0.078))
Error in sginv(cov(sm.prop), tol = .Machine$double.eps^(3/4)) : 
  Matrix a has negative elements on the diagonal, and snnd=TRUE.
```

Thanks,
Sam

**Comment by krivit** [comment]
The problem appears to be that `degcor` is undefined on an empty network, and its summary statistic ends up returning 0/0 or NaN. I guess we can treat an empty network as a special case that returns 0 by definition, or make it a parameter.

I am not sure it's a good idea to use it in modelling, either, since it lacks locality when its alternatives (`degcrossprod`?) don't.

**Comment by smjenness** [comment]
Thanks. Note that `degcrossprod` also generates the same error on an empty network. If we wanted to use it in the context of an egocentric/target stats approach, would we first simulate a random network with some mean degree, and then use that as the `nw` on the left hand side of the ergm forumula. Thanks for your help.

**Reply by krivit** [reply]
> Thanks. Note that `degcrossprod` also generates the same error on an empty network. If we wanted to use it in the context of an egocentric/target stats approach, would we first simulate a random network with some mean degree, and then use that as the `nw` on the left hand side of the ergm forumula. Thanks for your help.

Huh... I had forgotten that `degcrossprod` divides by the total edge count as the last step. I wonder what would happen if we didn't?
